pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -104,11 +104,12 @@ func (p Provider) Ignition(ctx context.Context, in clusterapi.IgnitionInput) ([]
 	defer cancel()
 
 	bucketName := gcp.GetBootstrapStorageName(in.InfraID)
-	bucketHandle, err := gcp.CreateBucketHandle(ctx, bucketName)
+	storageClient, err := gcp.NewStorageClient(context.Background())
```

Comment:
Rather than context.Background, can this use the ctx passed in to the function? 

Also, I feel like the `ctx` with timeout created a few lines above this is now too far away from where it is being used. Do we still want to be setting this timeout?